monitoring/uss_qualifier/scd/executor/runner.py

```diff
@@ -0,0 +1,105 @@
+            # TODO implement reporting
+            self.print_targets_state()
+            raise RuntimeError("[SCD] Error: Unable to identify the target managing flight {}".format(
+                step.inject_flight.name if 'inject_flight' in step else step.delete_flight.flight_name
+            ))
+
+        if 'inject_flight' in step:
+            print("[SCD]     Step: Inject flight {} to {}".format(step.inject_flight.name, target.name))
+            resp = target.inject_flight(step.inject_flight, dry=dry)
+            # TODO: Move evaluation at the end of the execution
+            TestRunner.evaluate_inject_flight_response(step.inject_flight, resp, dry=dry)
+        elif 'delete_flight' in step:
+            print("[SCD]     Step: Delete flight {} to {}".format(step.delete_flight.flight_name, target.name))
+            target.delete_flight(step.delete_flight.flight_name, dry=dry)
+        else:
+            print("[SCD] Warning: no action defined for step {}".format(step.name))
+
+    def get_managing_target(self, flight_name: str) -> typing.Optional[TestTarget]:
+        """Returns the managing target which created a flight"""
+        for role, target in self.targets.items():
+            if target.is_managing_flight(flight_name):
+                return target
+        return None
+
+    def get_target(self, step: TestStep) -> typing.Optional[TestTarget]:
+        """Returns the target which should be called in the TestStep"""
+        if 'inject_flight' in step:
+            return self.targets[step.inject_flight.injection_target.uss_role]
+        elif 'delete_flight' in step:
+            return self.get_managing_target(step.delete_flight.flight_name)
+        else:
+            raise NotImplementedError("Unsupported step. A Test Step shall contain either a inject_flight or a delete_flight object.")
+
+    # TODO: Use this method as a canvas to create findings and move the evaluation at the end of or outside the execution.
+    @staticmethod
+    def evaluate_inject_flight_response(attempt: FlightInjectionAttempt, resp: InjectFlightResponse, dry=False) -> bool:
+        if dry and resp.result == InjectFlightResult.DryRun:
+            print("[SCD]     Result: SKIP")
+            return True
+        if resp.result not in attempt.known_responses.acceptable_results:
+            print("[SCD]     Result: ERROR. Received {}, expected one of {}. Reason: {}".format(
+                resp.result,
+                attempt.known_responses.acceptable_results,
+                resp.get('notes', None))
+            )
+            return False
+        print("[SCD]     Result: SUCCESS")
+        return True
+
+    def print_targets_state(self):
+        print("[SCD] Targets States:")
+        for name, target in self.targets.items():
+            print(f"[SCD]   - {name}: {target.created_flight_ids}")
+
+    def print_test_plan(self):
+        """Runs the automated test in dry mode to validate and print the test plan."""
```

Comment:
If we keep this, I think it would benefit from some additional documentation.  What does it mean to print the test plan?  It seems like what the test executor does is going to depend on how the USSs respond, so what does it do if they're not responding?

Reply:
I removed it since the mocking approach seems more appropriate.